geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/metadata/TableBasedMetadata.scala

```diff
@@ -229,7 +229,18 @@ trait TableBasedMetadata[T] extends GeoMesaMetadata[T] with LazyLogging {
   }
 
   // checks that the table is already created, and creates it if not
-  def ensureTableExists(): Unit = tableExists.set(true, false, createTable())
+  def ensureTableExists(): Unit = {
+//    do not use compareAndSet because unsafe.compareAndSwapInt is called in compareAndSet，may make problem for jdk11
```

Comment:
hmm, `compareAndSet` doesn't work in jdk11? is there an equivalent function we could use? the problem with having a separate `get` and `set` calls is that multiple threads might end up trying to create the table if this method is called simultaneously, whereas `compareAndSet` ensures that only one thread will actually try to create the table. (also, we currently require jdk8, and there are likely to be other issues with using jdk11, so we could punt the jdk11 problems to later if we wanted)

Reply:
> hmm, `compareAndSet` doesn't work in jdk11? is there an equivalent function we could use? the problem with having a separate `get` and `set` calls is that multiple threads might end up trying to create the table if this method is called simultaneously, whereas `compareAndSet` ensures that only one thread will actually try to create the table. (also, we currently require jdk8, and there are likely to be other issues with using jdk11, so we could punt the jdk11 problems to later if we wanted)

unsafe method is not visible to developers by default.If I remember correctly, this will get an error. As your world ，it looks like I can change this to `compareAndSet`

Reply:
> hmm, `compareAndSet` doesn't work in jdk11? is there an equivalent function we could use? the problem with having a separate `get` and `set` calls is that multiple threads might end up trying to create the table if this method is called simultaneously, whereas `compareAndSet` ensures that only one thread will actually try to create the table. (also, we currently require jdk8, and there are likely to be other issues with using jdk11, so we could punt the jdk11 problems to later if we wanted)

ohhhhh,My mistake. I put the jdk8 code implementation into jdk11 to consider. In fact in jdk11 the implementation of compareAndSet has been modified to use the native method instead of the unsafe method, so I'm actually ok, I've changed the code.